paddle/fluid/operators/mkldnn/concat_mkldnn_op.cc

```diff
@@ -134,7 +182,65 @@ class ConcatMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
     auto concat = prim_creator.CreateConcatPrimitive(concat_pd, output, place);
     stream(stream::kind::eager).submit({concat}).wait();
 
-    output->set_mkldnn_prim_desc(concat_pd.dst_primitive_desc());
+    output->set_layout(DataLayout::kMKLDNN);
+    output->set_format(GetDstMemFormat(concat_pd));
+  }
+
+  void ComputeINT8(const paddle::framework::ExecutionContext& ctx) const {
+    auto multi_input = ctx.MultiInput<Tensor>("X");
+    EnforceLayouts(multi_input);
+    Tensor* output = ctx.Output<Tensor>("Out");
+    int64_t concat_axis = static_cast<int64_t>(ctx.Attr<int>("axis"));
+    auto& dev_ctx =
+        ctx.template device_context<paddle::platform::MKLDNNDeviceContext>();
+    auto place = GetCpuPlace(ctx);
+
+    memory::data_type dt =
+        paddle::framework::ToMKLDNNDataType(multi_input[0]->type());
+
+    ConcatPrimitiveFactory<T> prim_creator;
+    std::string key = CreateKey(ctx, multi_input, concat_axis, dt);
```

Comment:
Regarding the re-use mechanism in general - it'd be much easier to just store the ConcatPrimitiveFactory in the context instead of creating all the keys and storing primitives separately, while logically they create a single whole. It'd also make the code much shorter.

Reply:
Thanks for your proposal. The reason why we put the key generation out of ConcatPrimitiveFactory is to follow the similar pattern of design and implementation in other MKL-DNN FP32/INT8 kernels, e.g., Conv, Pooling, Quantize, and Dequantize.